oci_store.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratify
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"slices"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras-go/v2/errdef"
+	"oras.land/oras-go/v2/registry"
+)
+
+// OCIStore is a store implementation based on OCI image layout.
+type OCIStore struct {
+	name  string
+	store *oci.ReadOnlyStore
```

Comment:
Does this stand that we can still have multiple `AuthClient/AuthProvider`

Reply:
Please correct me if I'm wrong. This is not the only entry for the executor to get the artifact. And we would have eg. `RegistryStore` to implement those store apis with Auth functionality

Reply:
We will have a `RegistryStore` later in another PR. This PR is for a different purpose. That's why it is marked as related.